### DISCUSSION TITLE
Should an empty Frontmatter cause an error?

### DISCUSSION THREAD

**Discussion by seneca** [nuejs/nue#123] [Q&A]
If I add an empty frontmatter at the top of the document
---

---
 I get an error, presumably,  caused by the parser.
Should this really be an error or is this just a bug?
Personally, I think an empty front matters should be silently skipped by the parser as if wasn't there at all.

**Comment by tipiirai** [comment]
I don't get the error personally. I'm testing with [this page](https://github.com/nuejs/create-nue/blob/master/hotreload-demo/index.md). Tried without front matter and with empty front matter.

**Reply by seneca** [reply]
Well, that's strange. I've used skeleton starter template. And so, when I create a new xx.md page and leave empty front matter in there I get an error widow, although nothing is displayed there. Will double-check of course and come back with more info.

**Reply by tipiirai** [reply]
The skeleton app seems to work for me as well. Check the about page there, which does not have the front matter. Please also upgrade to the latest Nue: `bun install --global nuekit@latest`

**Reply by seneca** [reply]
Very strange indeed
I tried again with updated Bun and Nue and get the following error message in the console:
✓ Engine: Bun 1.0.17
✓ Serving site from ./.dist/dev
✓ http://localhost:8080/ 
✓ index.html 
✓ try.html 
✓ try.html 
 9 | export function parseMarkdown(raw, meta_only) {
10 |   const [_, front, md] = raw.split(/---+\n/)
11 |   const matter = front?.endsWith('---') ? front.slice(0, -3) : front
12 |   const meta = front ? yaml.load(matter) : {}
13 | 
14 |   if (!meta.title) {
            ^
TypeError: null is not an object (evaluating 'meta.title')
      at parseMarkdown (/Users/xxx/.bun/install/global/node_modules/nuekit/src/util.js:14:8)
      at /Users/xxx/.bun/install/global/node_modules/nuekit/src/nuekit.js:72:31
      at processTicksAndRejections (:61:77)

and the error window in the browser:
<img width="336" alt="Screenshot 2023-12-19 at 09 48 29" src="https://github.com/nuejs/nue/assets/53141/ab5c018c-19d4-4617-945e-4adca72ce186">

This is the content of my file:
---


---

[About us](/about/index.html)

# Hello there
This should work.

**Comment by tipiirai** [comment]
Please try with a new dev version: ` bun install nuekit@0.1.7-dev --global`

**Reply by seneca** [reply]
Yes, that works, thanks :-)
